src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -656,6 +656,13 @@ JNI_METHOD(void, pairDevice)
         JniByteArray jniCsrNonce(env, csrNonce);
         commissioningParams.SetCSRNonce(jniCsrNonce.byteSpan());
     }
+
+    commissioningParams.SetICDRegistrationStrategy(ICDRegistrationStrategy::kBeforeComplete);
```

Comment:
please move commissioningParams.SetICDRegistrationStrategy(ICDRegistrationStrategy::kBeforeComplete); under if (icdRegistrationInfo != nullptr)

icdRegistrationInfo is providing the raw symmetric key from applicatiion, kBeforeComplete depends on whether the key class is there or not

Reply:
[PairingCommand.kt](https://github.com/project-chip/connectedhomeip/pull/31055/files#diff-641253fadbdb691c0a899252479807f15948822a05e67debc95efa1ae880cae4)
You may create the new pairingCommand, the bottom line is to add new icdRegistrationInfo in test

Reply:
offline discussion, 
if that is icd device, and no key in ICDRegistrationInfo, then we locally generate it, it is fine
if that is icd device, and there is key in ICDRegistrationInfo, it is fine
if that is not icd device, the registration will be skipped